### PR TITLE
[BEAM-2309] implement VALUES

### PR DESCRIPTION
**Pull request by xumingming**
1. implement VALUES
2. add support for data type CHAR(to be able to test VALUES)

R: @XuMingmin

### CODE REVIEW

**Comment by xumingming** [comment]
Retest this please.

File: dsls/sql/src/main/java/org/apache/beam/dsls/sql/rel/BeamValuesRel.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+import org.apache.beam.sdk.transforms.Create;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.calcite.plan.RelOptCluster;
+import org.apache.calcite.plan.RelTraitSet;
+import org.apache.calcite.rel.core.Values;
+import org.apache.calcite.rel.type.RelDataType;
+import org.apache.calcite.rex.RexLiteral;
+
+/**
+ * {@code BeamRelNode} to replace a {@code Values} node.
```
Comment (XuMingmin) [inline dsls/sql/src/main/java/org/apache/beam/dsls/sql/rel/BeamValuesRel.java]: Can you add doc to describe the usage of `VALUES` in SQL.

File: dsls/sql/src/main/java/org/apache/beam/dsls/sql/rel/BeamValuesRel.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+  @Override public PCollection<BeamSQLRow> buildBeamPipeline(
+      BeamPipelineCreator planCreator) throws Exception {
+    List<BeamSQLRow> rows = new ArrayList<>(tuples.size());
+    String stageName = BeamSQLRelUtils.getStageName(this);
+    if (tuples.isEmpty()) {
+      throw new IllegalStateException("Values with empty tuples!");
+    }
+
+    BeamSQLRecordType beamSQLRecordType = deduceRecordType(tuples.get(0));
```
Comment (XuMingmin) [inline dsls/sql/src/main/java/org/apache/beam/dsls/sql/rel/BeamValuesRel.java]: can this be generated from `RelDataType rowType`?

File: dsls/sql/src/main/java/org/apache/beam/dsls/sql/rel/BeamValuesRel.java (hunk 1 above)
Comment (xumingming) [inline dsls/sql/src/main/java/org/apache/beam/dsls/sql/rel/BeamValuesRel.java]: Sure.

File: dsls/sql/src/main/java/org/apache/beam/dsls/sql/rel/BeamValuesRel.java (hunk 2 above)
Comment (xumingming) [inline dsls/sql/src/main/java/org/apache/beam/dsls/sql/rel/BeamValuesRel.java]: Very good catch! thanks

**Comment by xumingming** [comment]
@jbonofre can you take a look at this one?

**Comment by xumingming** [comment]
@jbonofre can you take a look at this one please?

**Comment by jbonofre** [comment]
R: @jbonofre

**Comment by jbonofre** [comment]
Merged on `DSL_SQL`. You can close this pull request. Thanks !

**Comment by xumingming** [comment]
Thanks @XuMingmin and @jbonofre !

**Pull request closed without merging**